paper/paper.md

```diff
@@ -29,52 +29,73 @@ bibliography: paper.bib
+-->
 
-There exist solutions to compute local minima of \eqref{eq:nlp} in Julia.
-A classical approach is to model the problem using `JuMP` [@jump], from the JuMP-dev organization, and then pass the model to state-of-the-art solvers via `MathOptInterface.jl` [@legat2021mathoptinterface], a thin wrapper to solvers that are typically available in C or Fortran.
-Another organization for nonlinear optimization is JuliaNLSolvers with `Optim.jl` [@mogensen2018optim], a package for univariate and multivariate optimization of functions. For instance, it implements in pure Julia an interior-point primal-dual Newton algorithm for solving \eqref{eq:nlp}.
-Finally, JSO also offers alternative solutions such as thin wrappers to the solvers Artelys Knitro [@byrd2006k] via `NLPModelsKnitro.jl` [@orban-siqueira-nlpmodelsknitro-2020] and Ipopt [@wachter2006implementation] via `NLPModelsIpopt.jl` [@orban-siqueira-nlpmodelsipopt-2020],
-but also `Percival.jl` [@percival-jl], a pure Julia implementation of an augmented Lagrangian penalization method.
-The main advantage of using JSO-compliant solvers is the flexibility in the origin of the inputted problem.
-To the best of our knowledge, there is no currently maintained available open-source implementation of DCI. Hence, we offer an interesting alternative to augmented Lagrangian or interior-point methods that are regularly used in the references mentioned above. 
+There already exist ways to solve \eqref{eq:nlp} in Julia.
+If \eqref{eq:nlp} is amenable to being modeled in `JuMP` [@jump], the model may be passed to state-of-the-art solvers, implemented in low-level compiled languages, via wrappers thanks to Julia's native interoperability with such languages.
+However, interfaces to low-level languages have limitations that pure Julia implementations do not have, including the ability to apply solvers with various arithmetic types.
+`Optim.jl` [@mogensen2018optim] implements a pure Julia primal-dual interior-point method for problems with both equality and inequality constrained modeled after Artlelys Knitro and Ipopt.
+
+JSO offers both types of solution mechanisms with thin wrappers to Artelys Knitro [@byrd2006k] via `NLPModelsKnitro.jl` [@orban-siqueira-nlpmodelsknitro-2020] and Ipopt [@wachter2006implementation] via `NLPModelsIpopt.jl` [@orban-siqueira-nlpmodelsipopt-2020] that let users pass in an `AbstractNLPModel`, and `Percival.jl` [@percival-jl], a pure Julia implementation of an augmented Lagrangian method based on bound-constrained subproblems.
+One main advantage of JSO-compliant solvers is the consistent API; the origin of the input problem is irrelevant.
+Finally, to the best of our knowledge, there is no available open-source implementation of DCI in existence. Hence, we offer an interesting alternative to augmented Lagrangian and interior-point methods in the form of an evolving, research level yet stable and mature, solver.
```

Comment:
Technically there is my old implementation.
```suggestion
Finally, to the best of our knowledge, there is no available maintained open-source implementation of DCI in existence. The original authors did not make their implementation public, and the other known implementation is `dcicpp` [@dcicpp], extending the original method to inequalities in the Ph.D. thesis by @siqueira2013controle, and it has had no updates in the last 5 years. Hence, we offer an interesting alternative to augmented Lagrangian and interior-point methods in the form of an evolving, research level yet stable and mature, solver.
```

Link to dcicpp: https://github.com/abelsiqueira/dcicpp

My thesis:
```
@phdthesis{siqueira2013controle,
title= {Controle din{\^a}mico de infactibilidade para programa{\c{c}}{\~a}o n{\~a}o linear},
author= {Soares Siqueira, Abel},
year= {2013},
school= {Universidade Estadual de Campinas}
}
```